### DISCUSSION TITLE
Best practice for merging several Multitomic datasets with snRNAs?

### DISCUSSION THREAD

**Discussion by GeorgiaZhao** [stuart-lab/signac#1012] [Q&A] [answered]
Dear Signac team,

I am trying to merge several Multitome datasets (snRNA+ATAC) with several snRNA datasets, all collected with 10X Genomic platform. Could you give me some help with the best way to integrate these datasets into one?

The detailed Vigenettes and posts on this topic are very helpful. To my best understanding, the integration requires several pre-processing steps includes:

1) create common peak coordinates across datasets (which I am planing to use MACS2 outputs bed files?)

2) merge all the fragments from all samples

After that, we can integrate ATAC with RNA data with a chosen references.

But when it comes to the actual practice, I still find several details are hard to resolve...

1) Should I merge all ATAC, and all RNAs independently and then merge the two? Following the line, how should I deal with the RNAs in the multitomic data that contains both ATACs and RNAs? Do I separate them into separated data set?

2) When I merge them, can the merge function know the cells that share the same names are the "same"? And along the same line, this naming system might become mismatched when merging ATAC and RNA separately?

3) I am not certain when to use "merge" and when to use "Integration". Should I "merge" ATAC and RNA separately, and then "integrate" the two or run things several times? OR I should simply dump everything into the "FindIntegrationAnchors" and "IntegrateEmbeddings" and wait for the magic to happen?

This is my first time working with ATAC data. Please forgive me if I make a simple problem complicated... It will be sincerely appreciated if you could give me some guidelines...

Thank you!

Georgia

**Comment by timoast** [comment, marked as answer]
To merge multiome dataset with snRNA datasets, this is what I'd suggest doing:

1. Quantify the same set of peaks across all the multiome datasets to be merged and create a new assay in each object, with the same name
2. Remove the original ATAC assay from each multiome object that contained the independent peak set for each object
3. Merge all the multiome and snRNA objects using the `merge()` function

Once this is done, you can process each assay (RNA and ATAC) as normal and check whether you see cells clustering based on their dataset of origin. If so, you can try doing integration. You can find integration anchors using the RNA assay since this is shared across all the datasets, following the same workflow shown on the Seurat website.

> When I merge them, can the merge function know the cells that share the same names are the "same"? And along the same line, this naming system might become mismatched when merging ATAC and RNA separately?

Cell name conflicts will be handled automatically

> merge all the fragments from all samples

There's no need to alter the fragment files

**Comment by GeorgiaZhao** [comment]
Hello,

Thank you for the clear guild! 

I can follow most of the instructions. But there is one thing I need a bit further clarification though...

When you say "Quantify the same set of peaks across all the multiome datasets to be merged", do you mean to follow the "merge object" vignettes showing [here](https://satijalab.org/signac/articles/merging.html)?

Thanks!

Georgia

**Reply by timoast** [reply]
yep

**Reply by GeorgiaZhao** [reply]
Sounds doable! I will give it a try.
Thanks,
Georgia

**Reply by pmf816** [reply]
Hello,
Georgia, may I ask your solutions? I also encountered the same issue. According to timoast's reply, I understand the following steps:

1. Merge multiple ATAC objects from the multitomic data into a new object (eg. atac.obj), base on common peaks;
2. Merge all snRNAs datasets into one object (eg. rna.obj);
4. Integrate ATAC and RNA data independently if cells cluster primarily by their dataset of origin;
3. Add atac.obj into rna.obj into a final big object.

Please point it out if I am wrong.

By the way, I don't quite understand "remove the original ATAC assist from each multiome object that contains the independent peak set for each object". How do I specifically operate?

I would appreciate it if anyone could provide me any suggestions or thoughts : D
